### ISSUE TITLE
Analyses for Paper

### ISSUE THREAD

**Issue by tsalo** [NBCLab/athena#36]
Here we can discuss any post-classification analyses @mriedel56 will perform for the upcoming paper. Feel free to edit this comment to add new analyses or respond to it with your thoughts.

We will run a separate nested CV for each combination of data source, feature space, dimension-wise feature boosting (i.e., yes or no), and classifier. The inner loop in the CV will be for hyper parameter tuning. The best parameters for that loop's model will be determined from the inner loop and a model will be trained on the outer loop's training data using those parameters. By performing this repeatedly, with shuffling, this method will result in ~300 (30 trials \* 10 outer folds) estimates of test error per combination. The distributions of test errors will be compared in the analyses.
## Comparison of Data Sources

Compare performance of classifiers using abstract, full text, methods section, and methods/abstract combination.
## Comparison of Classifiers

Compare performance of different classifiers. All classification will be done using BR within sklearn. Base learners will include:
- Bernoulli Naive Bayes
  - alpha: [0.01, 0.1, 1, 10]
- KNN
  - k: [1, 3, 5, 7, 9]
  - distance: [L1, L2]
- Logistic Regression
  - C: [0.01, 0.1, 1, 10, 100]
  - penalty: [L1, L2]
- Linear SVM
  - C: [0.01, 0.1, 1, 10, 100]
  - penalty: [L1, L2]
## Comparison of Feature Spaces

Compare performance of classifiers using different feature spaces and combinations of feature spaces. We'll only be including three feature spaces:
- Naive bag of words (NBOW)
- Cognitive Atlas
## Comparison of Dimension-wise Feature Boosting

Compare classifiers that boost dimension-wise classifier performance by predictions from other dimensions against those that don't.
- Feed the predictions from each dimension into the features for the next dimension, after ranking dimensions by either classifier performance (if the corpus from [the 2013 paper](http://journal.frontiersin.org/article/10.3389/fnins.2013.00240/full) is independent from this one) or qualitative "concreteness"/ease of classification.
## Possible Extensions
- Conditional F1-scores
  - Determine if failure to correctly classify certain labels is correlated with failures/successes of other labels.
- Effect of number of positive instances per label on classification accuracy
  - Determine if class imbalance or absolute number affect performance.
- Effect of proportion of experiments in paper with label on classification accuracy
  - Determine if papers that are "more" about a given label are more accurately classified.
